core/IO.carp

```diff
@@ -1,14 +1,78 @@
+; TODO: check if documentation needs parameters listed or if they are in function signature
+
 (system-include "carp_io.h")
 
 (register-type FILE)
 
 (defmodule IO
-  (doc      stdin "the standard input file (thin wrapper for the C standard library).")
-  (register stdin (Ptr FILE) "stdin")
-  (doc      stdout "the standard output file (thin wrapper for the C standard library).")
-  (register stdout (Ptr FILE) "stdout")
-  (doc      stderr "the standard error file (thin wrapper for the C standard library).")
-  (register stderr (Ptr FILE) "stderr")
+  
+  (doc Raw "wrappers for functions from the C standard library. Consider using a more carpesque function from IO when it exists. For detailed documentation please consult the documentation of your system (e.g. under Linux try man fprint).")
+  (defmodule Raw
+    (doc      stdin "the standard input file (thin wrapper for the C standard library).")
+    (register stdin (Ptr FILE) "stdin")
+    (doc      stdout "the standard output file (thin wrapper for the C standard library).")
+    (register stdout (Ptr FILE) "stdout")
+    (doc      stderr "the standard error file (thin wrapper for the C standard library).")
+    (register stderr (Ptr FILE) "stderr")
+    (doc      get-char "gets a character from stdin (thin wrapper for getchar() from C standard library).")
+    (register get-char (Fn [] Char) "getchar")  ; TODO: handle EOF?!
+    (doc      exit "exit the current program with a return code (thin wrapper for the C standard library).")
+    (register exit (Fn [Int] ()) "exit")  ; TODO: compare to System.exit for and remove 1 of them!
+    (doc      EOF "the End-Of-File character as a literal (thin wrapper for the C standard library)")
+    (register EOF Char "EOF")
+    (private  fopen-)
+    (hidden   fopen-)
+    (register fopen- (Fn [(Ptr CChar) (Ptr CChar)] (Ptr FILE)) "fopen" )
+    (doc      fopen "opens a file for input/output/appending (thin wrapper for fopen(pathname, mode) from C standard library). Consider using the function IO.open-file instead.")
+    (defn     fopen [pathname mode]
+      (fopen- (String.cstr pathname) (String.cstr mode)) )
+    (doc      fclose "closes a file pointer (thin wrapper for the C standard library).") 
+    (register fclose (Fn [(Ptr FILE)] Int)  "fclose")  
+    (doc      fclose! "same as (fclose) but to be used as a side effect.") 
+    (defn     fclose! [file]
+      (ignore (fclose file)) ) 
+    (private  fwrite-)
+    (hidden   fwrite-)
+    (register fwrite- (Fn [(Ptr CChar) Int Int (Ptr FILE)] Int) "fwrite")  
+    (doc      fwrite "writes a C-string to a file and returns the number of written items (thin wrapper for fwrite(cstr, item-size, items-count, file) from C standard library). Consider using [`write-file`](#write-file) instead.")
+    (defn     fwrite [data item-size items-count file]
```

Comment:
Should there be an `fwrite!` utility as well?

Reply:
In general it would make more sense to check whether anything was written (e.g. return value > 0) than whether a file was really closed.
Should I add it anyway?

Reply:
Done